app/models/admin/proposals/proposal.rb

```diff
@@ -140,12 +140,13 @@ def convert_to_show
     if Event.find_by(slug: @show.slug).present?
       p "Found a show with the same slug, which is #{@show.slug}."
       original_slug = @show.slug
-      number = 2
 
       # If so, append a number, and keep upping it until this Event's slug is unique.
-      while Event.find_by(slug: @show.slug).present? do
+      # Caps at 100. If it hits 100, something has gone very very wrong.
+      for number in 2..100 do
         @show.slug = "#{original_slug}-#{number}"
-        number += 1
+
+        break unless Event.find_by(slug: @show.slug).present?
```

Comment:
Maybe worth having some kind of error message to explain what's happened? Otherwise this could be pretty confusing if something does go amiss

Reply:
It will by default throw an error message that the conversion failed and will give as reason that the slug already exists. Is that clear enough?

Reply:
Since this is such weird behaviour I personally think maybe just having something to allude to what specifically has happened is worthwhile? But happy to be overruled on this